lib/matplotlib/tests/test_backend_tk.py

```diff
@@ -18,30 +20,32 @@ def _isolated_tk_test(success_count, func=None):
 
     TkAgg tests seem to have interactions between tests, so isolate each test
     in a subprocess. See GH#18261
-
-    The decorated function must be fully self-contained, and thus perform
-    all the imports it needs.  Because its source is extracted and run by
-    itself, coverage will consider it as not being run, so it should be marked
-    with ``# pragma: no cover``
     """
 
     if func is None:
         return functools.partial(_isolated_tk_test, success_count)
 
-    # Remove decorators.
-    source = re.search(r"(?ms)^def .*", inspect.getsource(func)).group(0)
-
+    if "MPL_TEST_ESCAPE_HATCH" in os.environ:
+        return func
```

Comment:
So, we are checking coverage of tests too? I guess I don't fully follow what coverage is reporting here...

Reply:
Yeah, we look at the coverage on the tests as well (@dopplershift has argued that tests are the only place insisting in 100% coverage actually make senses as if all of the code in your test suite is not running what are you even doing?!).

The change here is that instead of extracting the source and shoving it through `python -c ` we are importing the test module and then running the actual test function in a sub-process.  This bit of logic is so that when pytest discovers the tests it will go through the logic to spawn a sub-process to import an run the function but we import it to run it we need to actually run it.  Setting an env in the subprocess call was the simplest way I found to do this.

Reply:
And I new see the warning that this is not actually covered and I am confused....

Reply:
> (@dopplershift has argued that tests are the only place insisting in 100% coverage actually make senses as if all of the code in your test suite is not running what are you even doing?!).

There can be many reasons you don't run all of your test suite at a given time. Some optional dependencies may not be available, you could have very costly tests that you run only at larger intervals, ...

Out of curiosity, does coverage stretch across all the running CI jobs or where does it get its data from?

Reply:
yeah, at the end of all of our CI jobs we have something like

```yaml
      - name: Upload code coverage
        uses: codecov/codecov-action@v2
```

that pushes coverage information to codecov and then they merge all of them together. 

If you click through the coverage UI enough you can get to https://codecov.io/gh/matplotlib/matplotlib/commit/153bfa18377c1a15a76906cfb77973971e0cee95/build which will show you which CI runs contributed to a given report.